### PR TITLE
Ogc 947 show events in the past from link

### PR DESCRIPTION
**Pull request by Tschuppi81**
## Commit message

Events: Events from the past can be shown.

In the case of 'No events found' we will display two link buttons. One will point to 'all events' where as the other will show 'past events'. In addition a new time range filter 'past events' was added to achieve the same from filters.

TYPE: Feature
LINK: OGC-947

### CODE REVIEW

**Review by msom — COMMENTED** [review]
Looks good, a few small things:
- Use EN on github
- Update macro for org
- Update JS for date range filter
- Optionally hide "All Events" link if no filter is set

File: src/onegov/town6/templates/macros.pt [hunk 1]
```diff
@@ -1163,6 +1163,13 @@
     </tal:b>
         </div>
     <div tal:condition="not occurrences" i18n:translate>No events found.</div>
+    <div tal:condition="not occurrences">
+        <!-- "${layout.events_url}" -->
```
[the remark this answers is not in the document]
Comment (Tschuppi81) [reply src/onegov/town6/templates/macros.pt]: done

File: src/onegov/event/collections/occurrences.py [hunk 2]
```diff
@@ -116,6 +117,10 @@ def range_to_dates(self, range, start=None, end=None):
         if range == 'month':
             start = today.replace(day=1)
             return start, start + relativedelta(months=1, days=-1)
+        if range == 'past':
+            milenium = date(2000, 1, 1)
```
Comment (BreathingFlesh) [inline src/onegov/event/collections/occurrences.py:121]: "millenium" should have two L's, but it won't appear anywhere, so it doesn't really matter.

File: src/onegov/town6/templates/macros.pt [hunk 3]
```diff
@@ -1163,6 +1163,15 @@
     </tal:b>
         </div>
     <div tal:condition="not occurrences" i18n:translate>No events found.</div>
+    <div tal:condition="not occurrences">
+        <br>
+        <div tal:condition="'range' in layout.og_url">
+            <a href="${layout.events_url}" class="button hollow">Alle Veranstaltungen</a>
+        </div>
+        <div>
```
Comment (BreathingFlesh) [inline src/onegov/town6/templates/macros.pt:1171]: I don't think that div does anything.

**Review by BreathingFlesh — COMMENTED** [review]
Only minor changes. Comments on macros.pt in town6 also apply for macros in org.

File: src/onegov/town6/templates/macros.pt (hunk 3 above)
Comment (Tschuppi81) [reply src/onegov/town6/templates/macros.pt:1171]: done

File: src/onegov/town6/templates/macros.pt [hunk 4]
```diff
@@ -1163,6 +1163,15 @@
     </tal:b>
         </div>
     <div tal:condition="not occurrences" i18n:translate>No events found.</div>
+    <div tal:condition="not occurrences">
```
[the remark this answers is not in the document]
Comment (Tschuppi81) [reply src/onegov/town6/templates/macros.pt:1166]: good input

**Pull request merged**